clippy_lints/src/casts/mod.rs

```diff
@@ -454,6 +455,24 @@ declare_clippy_lint! {
     "casting using `as` between raw pointers to slices of types with different sizes"
 }
 
+declare_clippy_lint! {
+    /// ### What it does
+    /// Checks for casts from an enum tuple constructor to an integer.
+    ///
+    /// ### Why is this bad?
+    /// The cast is easily confused with casting a c-like enum value to an integer.
+    ///
+    /// ### Example
+    /// ```rust
+    /// enum E { X(i32) };
+    /// let _ = E::X as usize;
+    /// ```
+    #[clippy::version = "1.61.0"]
+    pub CAST_ENUM_CONSTRUCTOR,
+    restriction,
```

Comment:
should this be correctness? It seems rather rare for this to be legitimate and in those cases it's probably better served by casting to `fn` first

Reply:
Could also be in `suspicious`. It's not inherently wrong, but I haven't ever seen a case where anyone would actually want to do this so it could be either.

Reply:
suspicious works for me!

Reply:
Done.